### ISSUE TITLE
Telemetry handler experience

### ISSUE THREAD

**Issue by isvargasmsft** [microsoftgraph/msgraph-sdk-php#1209]
The telemetry handler would only set the client request Id if not already present, allowing people to set it manually if they need some kind of end-to-end tracing.

**Comment by isvargasmsft** [comment]
CC: @Ndiritu @SilasKenneth @shemogumbe

**Comment by Ndiritu** [comment]
For context, our telemetry handler currently adds 2 headers to requests: `SdkVersion` and `client-request-id`. Client Request ID can be manually set or defaults to a random UUID.

The current dev experience of customising client request ID when using service libraries:
```php

$telemetryOption = new GraphTelemetryOption();
$telemetryOption->setClientRequestId('custom id');

$requestConfig = new UsersRequestBuilderGetRequestConfiguration();
$requestConfig->options = [$telemetryOption];

$response = $graphServiceClient->users()->get($requestConfig)->wait();
```

Dev exp. when using Graph core only:
```php

$telemetryOption = new GraphTelemetryOption();
$telemetryOption->setClientRequestId('custom id');

$guzzleClient = GraphClientFactory::setTelemetryOption($telemetryOption)::create();
$response = $guzzleClient->get('/users');
```

Some improvements/gaps in the dev experience:
- can be shortened so that client request Id is set in the `GraphTelemetryOption` constructor
- current design could allow devs to overwrite SDK versions. We can find a way to "hide" those.

**Comment by isvargasmsft** [comment]
@baywet I understand you have more context on these 2 requests, is that right? These came from Seb.

**Comment by baywet** [comment]
@Ndiritu Open Telemetry is not yet implemented in PHP as far as I know, correct?

**Comment by Ndiritu** [comment]
Ah, didn't know we're referring that. 
Not yet.

**Comment by baywet** [comment]
right, so the second requirement can only be implemented when the tracing work is done. I suggest that you only implement the first now, and leave a note on the issue capturing tracing for the second.

**Comment by isvargasmsft** [comment]
Thank you very much @baywet

**Comment by baywet** [comment]
here is the related issue I'm referring to https://github.com/microsoft/kiota/issues/1871
